### PR TITLE
Always use Snowflakes in args

### PR DESCRIPTION
**Pull request by SpecificProtagonist**
Covers function arguments missed by #122. This also lead to an elimination of a bunch of clone calls.

### CODE REVIEW

File: src/api/channels/channels.rs [hunk 1]
```diff
@@ -43,12 +43,12 @@ impl Channel {
     /// # Returns
     ///
     /// A `Result` that contains a `ChorusLibError` if an error occurred during the request, or `()` if the request was successful.
-    pub async fn delete(self, user: &mut UserMeta) -> ChorusResult<()> {
+    pub async fn delete(channel_id: Snowflake, user: &mut UserMeta) -> ChorusResult<()> {
```
Comment (bitfl0wer) [inline src/api/channels/channels.rs:46]: Taking ownership of `self` was being done on purpose, since it disposes (drops) the channel after it has been deleted. Why change this?

Comment (bitfl0wer) [reply src/api/channels/channels.rs:46]: I'd say that if someone wants to delete a channel and still keep it in memory, the call to `clone()` is not much effort and relatively inexpensive in the grand scheme of things (since you are not expected to perform this operation a gazillion times). When I think about the Polyphony client, which is the main reason Chorus exists, I don't see why we would keep a Channel in memory after deleting it.

The method in which the channel is owned might not go out of scope for a long time, and calling drop() after deleting the channel manually feels a little silly. So I am not quite sure; From my POV, both your and my way of thinking make sense. I am not as experienced of a Rust developer though :P

Comment (bitfl0wer) [reply src/api/channels/channels.rs:46]: @kozabrada123 @Zert3x I'd love another opinion on this :)

Comment (Zert3x) [reply src/api/channels/channels.rs:46]: The way I handle this in my projects is to take ownership of the object if you are deleting it.  Since if you're deleting it, you no longer need it in any capacity.  If you do for some reason need the whole or part of the object after it's been deleted, then it would make sense to clone out that information.

Comment (bitfl0wer) [inline src/api/channels/channels.rs:46]: Please revert the change to the functions signature, the rest looks good! Thanks once again for this PR :)

**Review by bitfl0wer — CHANGES_REQUESTED** [review]

**Review by bitfl0wer — APPROVED** [review]
